src/libraries/Microsoft.PowerFx.Core/Public/Values/UntypedObjectValue.cs

```diff
@@ -106,12 +106,23 @@ public abstract class UntypedObjectBase : IUntypedObject
         /// <summary>
         /// Set a property on the object.
         /// </summary>
-        /// <param name="propertyName">Property name.</param>
+        /// <param name="propertyName">Property identifier.</param>
         /// <param name="value">FormulaValue to be set.</param>
         public virtual void SetProperty(string propertyName, FormulaValue value)
         {
             // In case of unwanted behavior, throw an CustomFunctionErrorException exception.
             throw new NotImplementedException();
         }
+
+        /// <summary>
+        /// Set a property on the object.
+        /// </summary>
+        /// <param name="index">Property identifier.</param>
+        /// <param name="value">FormulaValue to be set.</param>
+        public virtual void SetProperty(int index, FormulaValue value)

```

Comment:
>index [](http://example.com/codeflow?start=44&length=5)

we should comment whether this is 0-based or 1-based.

Reply:
Isn't this up to the host to decide?

Reply:
Or simpler .. what does this return? 
First(ParseJSON("[10,20,30]"))